xtask/src/codegen/diagnostics_docs.rs

```diff
@@ -9,14 +9,16 @@ use crate::{
 };
 
 pub(crate) fn generate(check: bool) {
-    let diagnostics = Diagnostic::collect().unwrap();
-    if !check {
-        let contents =
-            diagnostics.into_iter().map(|it| it.to_string()).collect::<Vec<_>>().join("\n\n");
-        let contents = add_preamble(crate::flags::CodegenType::DiagnosticsDocs, contents);
-        let dst = project_root().join("docs/book/src/diagnostics_generated.md");
-        fs::write(dst, contents).unwrap();
+    if check {
+        return;
     }
+
+    let diagnostics = Diagnostic::collect().unwrap();
+    let contents =
+        diagnostics.into_iter().map(|it| it.to_string()).collect::<Vec<_>>().join("\n\n");
+    let contents = add_preamble(crate::flags::CodegenType::DiagnosticsDocs, contents);
+    let dst = project_root().join("docs/book/src/diagnostics_generated.md");
+    fs::write(dst, contents).unwrap();
```

Comment:
Why this change? `let diagnostics = Diagnostic::collect().unwrap();` should run even if a check is requested

Reply:
Since the value was used only for the generation, I assumed we can ignore it when returning early. Fixed.